src/Series.cpp

```diff
@@ -85,145 +97,105 @@ Series::Series(std::string filepath,
         filepath = auxiliary::replace_all(filepath, "\\", "/");
     }
 #endif
-    std::string newPath;
-    std::string newName;
     auto const pos = filepath.find_last_of(auxiliary::directory_separator);
     if( std::string::npos == pos )
     {
-        newPath = ".";
-        newPath.append(1, auxiliary::directory_separator);
-        newName = filepath;
+        input->path = ".";
+        input->path.append(1, auxiliary::directory_separator);
+        input->name = filepath;
     }
     else
     {
-        newPath = filepath.substr(0, pos + 1);
-        newName = filepath.substr(pos + 1);
+        input->path = filepath.substr(0, pos + 1);
+        input->name = filepath.substr(pos + 1);
     }
 
-    IterationEncoding ie;
-    if( std::string::npos != newName.find("%T") )
-        ie = IterationEncoding::fileBased;
+    input->format = determineFormat(input->name);
+
+#if defined(IS_GCC_48)
+    regex_t pattern;
+    if( regcomp(&pattern, "(.*)%(0\\d+)?T(.*)", REG_EXTENDED) )
+        throw std::runtime_error(std::string("Regex for iterationFormat '(.*)%(0\\d+)?T(.*)' can not be compiled!"));
+
+    uint8_t maxGroups = 4;
```

Comment:
make this a constexpr to avoid declaring a variable length array

Reply:
I'll probably opt for an ```std::array< regmatch_t, 4 >```. This is C++, after all.

Reply:
I still would keep the named constant as a separate line. Magic numbers are evil.
```C++
constexpr uint8_t maxGroups = 4;
std::array< regmatch_t, maxGroups > regexMatch;
```